src/dispatchers/networkDispatchers.ts

```diff
@@ -173,6 +173,11 @@ export class FetchRequestDispatcher extends Dispatcher<FetchRequest, channels.Fe
     super(scope, request, 'FetchRequest', {}, true);
   }
 
+  async dispose(params?: channels.FetchRequestDisposeParams): Promise<void> {
+    this._object.dispose();
```

Comment:
I am a bit confused why are these two different?
- For example, I'd put `this._object.dispose()` directly into `_disposeDispatcher` - will that work?
- Alternatively, I'd only call `this._object.dispose()` everywhere, and emit a "disposed" event that dispatcher would listen to (similarly to context's "closed" event).

Reply:
> * For example, I'd put `this._object.dispose()` directly into `_disposeDispatcher` - will that work?

That works but it is indirect, i.e. FetchRequest is created by BrowserContext directly, owned by it but then disposed by an external dispatcher. This boils down to the same problem of separation between dispatchers and model objects, other than that disposing FetchRequest along with its dispatcher would work.

> * Alternatively, I'd only call `this._object.dispose()` everywhere, and emit a "disposed" event that dispatcher would listen to (similarly to context's "closed" event).

By everywhere you mean in FetchRequestDispatcher.dispose and BrowserContext._didCloseInternal ?